adapters/oidc/installed/src/main/java/org/keycloak/adapters/installed/KeycloakInstalled.java

```diff
@@ -289,6 +302,86 @@ public void loginManual(PrintStream printer, Reader reader) throws IOException,
         status = Status.LOGGED_MANUAL;
     }
 
+    public boolean loginCommandLine() throws IOException, ServerRequest.HttpFailure, VerificationException {
+        String redirectUri = "urn:ietf:wg:oauth:2.0:oob";
+
+        return loginCommandLine(redirectUri);
+    }
+
+
+
+    /**
+     * Proprietary WWW-Authentication challenge protocol.
+     * WWW-Authentication: X-Text-Form-Challenge callback="{url}" param="{param-name}" label="{param-display-label}"
+     *
+     * @param redirectUri
+     * @return
+     * @throws IOException
+     * @throws ServerRequest.HttpFailure
+     * @throws VerificationException
+     */
+    public boolean loginCommandLine(String redirectUri) throws IOException, ServerRequest.HttpFailure, VerificationException {
+        String authUrl = deployment.getAuthUrl().clone()
+                .queryParam(OAuth2Constants.RESPONSE_TYPE, OAuth2Constants.CODE)
+                .queryParam(OAuth2Constants.CLIENT_ID, deployment.getResourceName())
+                .queryParam(OAuth2Constants.REDIRECT_URI, redirectUri)
+                .queryParam(OAuth2Constants.SCOPE, OAuth2Constants.SCOPE_OPENID)
+                .build().toString();
+        ResteasyClient client = new ResteasyClientBuilder().disableTrustManager().build();
```

Comment:
Doesn't look very safe? Couldn't it reuse the ResteasyClient based on the Apache HTTP Client from the provided KeycloakDeployment, which has all the truststore related stuff setup?

Reply:
KeycloakInstalled is not part of the server.  This is an experimental feature anyways.

Reply:
experemental and optional feature.